13/umbraco-cms/reference/configuration/imagingsettings.md

```diff
@@ -82,3 +83,70 @@ public class ConfigureImageSharpMiddlewareOptionsComposer : IComposer
         });
 }
 ```
+
+## Hmac secret key
+
+Specifies the key used to secure image requests by generating a hash-based message authentication code (HMAC). This ensures that only valid requests can access or manipulate images.
+
+To enable it, you need to set a secure random key. This key should be kept secret and not shared publicly. The key can be set through the IOptions pattern, or you can insert a base64 encoded key in the `appsettings.json` file. The key should ideally be 64 bytes long.
+
+The key must be the same across all environments (development, staging, production) to ensure that image requests work for content published across environments.
+
+### Default Behavior
+
+If the `HMACSecretKey` is not set, image requests will not be secured, and any person can request images with any parameters. This may expose your server to abuse, such as excessive resizing requests or unauthorized access to images.
+
+### Key Length
+
+The `HMACSecretKey` should be a secure, random key. For most use cases, a 64-byte (512-bit) key is recommended. If you are using HMACSHA384 or HMACSHA512, you may want to use a longer key (e.g., 128 bytes).
+
+### Example Configuration
+
+**Appsettings.json**
+```json
+"Umbraco": {
+  "CMS": {
+    "Imaging": {
+      "HMACSecretKey": "some-base64-encoded-key"
+    }
+  }
+}
+```
+
+**Using the IOptions pattern**
+
+If you prefer to generate the `HMACSecretKey` programmatically or want to avoid hardcoding it in your configuration files, you can use the `IOptions` pattern. The following example demonstrates how to generate a secure random key at runtime:
```

Comment:
```suggestion
To generate the `HMACSecretKey` programmatically instead of hardcoding it in configuration files, use the `IOptions` pattern. The following example demonstrates how to generate a secure random key at runtime:
```